### ISSUE TITLE
LocalStack fails to start container with docker v20

### ISSUE THREAD

**Issue by xmlprgrm** [localstack/localstack#3446]
# Type of request: This is a ...

[x ] bug report
[ ] feature request

# Detailed description
I am trying to run my unit tests with testcontainers. I am using 
- Mac catalina
- testcontainers version 1.15.1
- localstack image v0.12.4. 
- Docker version 20.10.0
...

## Expected behavior

...

## Actual behavior
When I run the tests I dont see the container starting. There is a log statement "Creating container" and then it hangs. 

I cannot repro the same with docker engine v19. Started seeing it only after I upgraded. Can anyone please tell me what is wrong?
...

# Steps to reproduce

## Command used to start LocalStack

...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

...

**Comment by rmoak** [comment]
I am seeing similar behavior. I put together a sample Java project that should repro the issue - https://github.com/rmoak/localstack-docker-error

It works fine on Docker For Mac v2.5.0.1 with Engine 19.03.13 but on Docker For Mac 3.0.0 with Engine v20.10.0, the script will run properly once and then each subsequent run will hang at the `[main] INFO 🐳 [localstack/localstack:latest] - Creating container for image: localstack/localstack:latest` step.

After it hangs, `docker ps -a` shows:
```
7fc8cc2c099a   alpine:3.5                  "/bin/sh"   36 seconds ago      Created                                   boring_leavitt
1a1bde65adae   testcontainers/ryuk:0.3.0   "/app"      37 seconds ago      Up 36 seconds   0.0.0.0:55004->8080/tcp   testcontainers-ryuk-9c87a286-cdfa-4783-b09c-00808126cd6b
```

And `docker inspect 7fc8cc2c099a`, `docker stats 7fc8cc2c099a` and `docker kill 7fc8cc2c099a` all hang. 

After exitting the java process launched, the two containers above will remain with the `alpine:3.5` container still unreachable. Restarting the Docker For Mac v3 process makes them reachable again and will allow the above java app to launch again but then subsequent runs re-exhibit the hung behavior until another restart of Docker is performed.

I am not sure if this is actually specifically a Localstack issue but piling on here as this is where I am currently facing the issue.

**Comment by mgagliardo** [comment]
Hello @rmoak @xmlprgrm, thanks for reporting this.

First of all, I have the following setup:
```
MacOS Big Sur v11.1 (20C69)
Docker version 20.10.2, build 2291f61 
Docker desktop 3.1.0 (51484)
Docker image localstack/localstack:latest
```

I have pulled @rmoak 's repo and (just one small fix, I had to add the maven plugins to pom.xml) specifying Java 11:
```
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.1</version>
                <configuration>
                    <release>11</release>
                </configuration>
            </plugin>
            <plugin>
```

Anyways, I was unable to reproduce this issue under these conditions, this is the 5th time I ran this:
```
$ java -jar target/localstack-error-1.0-SNAPSHOT-jar-with-dependencies.jar
[main] INFO org.testcontainers.dockerclient.DockerClientProviderStrategy - Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
[main] INFO org.testcontainers.dockerclient.DockerClientProviderStrategy - Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
[main] INFO org.testcontainers.DockerClientFactory - Docker host IP address is localhost
[main] INFO org.testcontainers.DockerClientFactory - Connected to docker:
  Server Version: 20.10.2
  API Version: 1.41
  Operating System: Docker Desktop
  Total Memory: 1990 MB
[main] INFO org.testcontainers.utility.ImageNameSubstitutor - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
[main] INFO org.testcontainers.DockerClientFactory - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
[main] INFO org.testcontainers.DockerClientFactory - Checking the system...
[main] INFO org.testcontainers.DockerClientFactory - ✔︎ Docker server version should be at least 1.6.0
[main] INFO org.testcontainers.DockerClientFactory - ✔︎ Docker environment should have more than 2GB free disk space
[main] INFO 🐳 [localstack/localstack:latest] - HOSTNAME_EXTERNAL environment variable set to localhost (to match host-routable address for container)
[main] INFO 🐳 [localstack/localstack:latest] - Creating container for image: localstack/localstack:latest
[main] INFO 🐳 [localstack/localstack:latest] - Starting container with ID: 19ac9c7644375904237e93c7dd00c4169d83a982fd62073ae9274d2207511fc2
[main] INFO 🐳 [localstack/localstack:latest] - Container localstack/localstack:latest is starting: 19ac9c7644375904237e93c7dd00c4169d83a982fd62073ae9274d2207511fc2
[main] INFO 🐳 [localstack/localstack:latest] - Container localstack/localstack:latest started in PT13.079996S
```

This is what I've seen in my `docker ps-a`:
```
Every 2.0s: docker ps -a                                                                                                                              Miguel-Angel-Gagliardo.local: Sat Jan 23 21:40:25 2021

CONTAINER ID   IMAGE                          COMMAND                  CREATED          STATUS          PORTS                                                                       NAMES
3762bd88ac65   localstack/localstack:latest   "docker-entrypoint.sh"   10 seconds ago   Up 9 seconds    0.0.0.0:55019->4566/tcp, 0.0.0.0:55018->4571/tcp, 0.0.0.0:55017->8080/tcp   blissful_goodall
5cd6fd4951fb   testcontainers/ryuk:0.3.0      "/app"                   11 seconds ago   Up 10 seconds   0.0.0.0:55016->8080/tcp                                                     testcontainers-ryuk-44bb
3936-de04-403b-a563-0360a76d08c8
```

Can you please point me if I am doing something wrong or how can I reproduce this? I was unable to do so, or maybe is my setup..

Thanks.

**Comment by rmoak** [comment]
Hm. I am running OSX Mojave 10.14.6 and Open JDK 1.8.0_265. Let me try updating to see if that repro's and I will report back. Thanks for looking into this.

**Comment by mgagliardo** [comment]
@rmoak any updates on this?

**Comment by rmoak** [comment]
Sorry for the delay. Still working on it - Big Sur has not been greenlit for deployment just yet by internal IT policies. I have a request in.

**Comment by ryanmoaksonos** [comment]
@mgagliardo I was still able to repro on Big Sur **but I believe I figured out the issue**. 

TL;DR - It seems the experimental feature `Use gRPC FUSE for file sharing` is the issue. Having it disabled causes follow up runs to fail. 

Here is my writeup that I would have posted before I thought to try toggling this setting.

My current env should match yours:
```
MacOS Big Sur v11.1 (20C69)
Docker version 20.10.2, build 2291f61 
Docker desktop 3.1.0 (51484)
Docker image localstack/localstack:latest
openjdk version "15.0.2"
```

On the first run I see the exact same log output as you did. With the same two containers left behind and cleaned up shortly after the JVM exists. 


Now on each run after that, it seems to always hang when starting the alpine:3.5 container. Running `java -jar -Dorg.slf4j.simpleLogger.defaultLogLevel=DEBUG target/localstack-error-1.0-SNAPSHOT-jar-with-dependencies.jar` to get better DEBUG info I see it hang here:

```
[main] INFO 🐳 [localstack/localstack:latest] - HOSTNAME_EXTERNAL environment variable set to localhost (to match host-routable address for container)
[main] DEBUG 🐳 [localstack/localstack:latest] - Starting container: localstack/localstack:latest
[main] DEBUG 🐳 [localstack/localstack:latest] - Trying to start container: localstack/localstack:latest (attempt 1/1)
[main] DEBUG 🐳 [localstack/localstack:latest] - Starting container: localstack/localstack:latest
[main] INFO 🐳 [localstack/localstack:latest] - Creating container for image: localstack/localstack:latest
[main] DEBUG org.testcontainers.utility.RegistryAuthLocator - Looking up auth config for image: localstack/localstack:latest at registry: index.docker.io
[main] DEBUG org.testcontainers.utility.RegistryAuthLocator - Cached auth found: [AuthConfig{username=rmoak, password=hidden non-blank value, auth=blank, email=null, registryAddress=index.docker.io, registryToken=blank}]
[main] DEBUG org.testcontainers.dockerclient.AuthDelegatingDockerClientConfig - Effective auth config [AuthConfig{username=rmoak, password=hidden non-blank value, auth=blank, email=null, registryAddress=index.docker.io, registryToken=blank}]
[main] DEBUG org.testcontainers.utility.PrefixingImageNameSubstitutor - No prefix is configured
[main] DEBUG org.testcontainers.utility.ImageNameSubstitutor - Did not find a substitute image for alpine:3.5 (using image substitutor: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor'))
[main] DEBUG org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd - Cmd: alpine:3.5
[main] DEBUG org.testcontainers.utility.RegistryAuthLocator - Looking up auth config for image: alpine:3.5 at registry: index.docker.io
[main] DEBUG org.testcontainers.utility.RegistryAuthLocator - Cached auth found: [AuthConfig{username=rmoak, password=hidden non-blank value, auth=blank, email=null, registryAddress=index.docker.io, registryToken=blank}]
[main] DEBUG org.testcontainers.dockerclient.AuthDelegatingDockerClientConfig - Effective auth config [AuthConfig{username=rmoak, password=hidden non-blank value, auth=blank, email=null, registryAddress=index.docker.io, registryToken=blank}]
[main] DEBUG org.testcontainers.utility.MountableFile - Copying classpath resource(s) from jar:file:/Users/ryan.moak/git/localstackerror/target/localstack-error-1.0-SNAPSHOT-jar-with-dependencies.jar!/org/testcontainers/utility/ResourceReaper.class to /tmp/.testcontainers-tmp-11480015398599898168 to permit Docker to bind
[main] DEBUG org.testcontainers.utility.MountableFile - Copying resource org/testcontainers/utility/ResourceReaper.class from JAR file /Users/ryan.moak/git/localstackerror/target/localstack-error-1.0-SNAPSHOT-jar-with-dependencies.jar
[main] DEBUG org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd - Cmd: org.testcontainers.shaded.com.github.dockerjava.core.command.CreateContainerCmdImpl@4b86805d[name=<null>,hostName=<null>,domainName=<null>,user=<null>,attachStdin=<null>,attachStdout=<null>,attachStderr=<null>,portSpecs=<null>,tty=<null>,stdinOpen=<null>,stdInOnce=<null>,env=<null>,cmd=<null>,healthcheck=<null>,argsEscaped=<null>,entrypoint=<null>,image=alpine:3.5,volumes=com.github.dockerjava.api.model.Volumes@5852c06f,workingDir=<null>,macAddress=<null>,onBuild=<null>,networkDisabled=<null>,exposedPorts=com.github.dockerjava.api.model.ExposedPorts@4149c063,stopSignal=<null>,stopTimeout=<null>,hostConfig=HostConfig(binds=[/private/tmp/.testcontainers-tmp-11480015398599898168:/dummy:ro], blkioWeight=null, blkioWeightDevice=null, blkioDeviceReadBps=null, blkioDeviceWriteBps=null, blkioDeviceReadIOps=null, blkioDeviceWriteIOps=null, memorySwappiness=null, nanoCPUs=null, capAdd=null, capDrop=null, containerIDFile=null, cpuPeriod=null, cpuRealtimePeriod=null, cpuRealtimeRuntime=null, cpuShares=null, cpuQuota=null, cpusetCpus=null, cpusetMems=null, devices=null, deviceCgroupRules=null, deviceRequests=null, diskQuota=null, dns=null, dnsOptions=null, dnsSearch=null, extraHosts=null, groupAdd=null, ipcMode=null, cgroup=null, links=[], logConfig=LogConfig(type=null, config=null), lxcConf=null, memory=null, memorySwap=null, memoryReservation=null, kernelMemory=null, networkMode=null, oomKillDisable=null, init=null, autoRemove=null, oomScoreAdj=null, portBindings=null, privileged=null, publishAllPorts=null, readonlyRootfs=null, restartPolicy=null, ulimits=null, cpuCount=null, cpuPercent=null, ioMaximumIOps=null, ioMaximumBandwidth=null, volumesFrom=null, mounts=null, pidMode=null, isolation=null, securityOpts=null, storageOpt=null, cgroupParent=null, volumeDriver=null, shmSize=null, pidsLimit=null, runtime=null, tmpFs=null, utSMode=null, usernsMode=null, sysctls=null, consoleSize=null),labels={org.testcontainers=true, org.testcontainers.sessionId=31b219de-cbb5-4bf6-863e-a06645161b00},shell=<null>,networkingConfig=<null>,ipv4Address=<null>,ipv6Address=<null>,aliases=<null>,authConfig=AuthConfig(username=rmoak, email=null, registryAddress=index.docker.io, stackOrchestrator=null)]
[main] DEBUG org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd - Cmd: c99913e891d667c09cace1626941a7e333f7b7cfc6bcfbee103d42344684f255
```
Typically, this will just hang here forever. If I `CMD+C` out of the process and try to run again, it will hang at the same point but with the newly launch container Id. After continuing to try this a few times then once again - my `docker ps -a` will all look like this: 

```
ryan.moak:~$ docker ps -a
CONTAINER ID   IMAGE                       COMMAND     CREATED          STATUS          PORTS                     NAMES
c99913e891d6   alpine:3.5                  "/bin/sh"   4 seconds ago    Created                                   xenodochial_spence
f02332255b9d   testcontainers/ryuk:0.3.0   "/app"      5 seconds ago    Up 4 seconds    0.0.0.0:55007->8080/tcp   testcontainers-ryuk-31b219de-cbb5-4bf6-863e-a06645161b00
137fdd9048fb   alpine:3.5                  "/bin/sh"   12 seconds ago   Created                                   blissful_carson
efcb33ce8c96   testcontainers/ryuk:0.3.0   "/app"      13 seconds ago   Up 12 seconds   0.0.0.0:55006->8080/tcp   testcontainers-ryuk-7effc270-09fb-44c2-9b7e-92668992a5a2
c8cc783a9c9a   alpine:3.5                  "/bin/sh"   8 minutes ago    Created                                   angry_allen
5b0594f80df5   testcontainers/ryuk:0.3.0   "/app"      8 minutes ago    Up 8 minutes    0.0.0.0:55005->8080/tcp   testcontainers-ryuk-2463326d-a198-47fb-aa89-2178da8d1b44
a530d3802362   alpine:3.5                  "/bin/sh"   10 minutes ago   Created                                   eloquent_turing
41a67f68ed9c   testcontainers/ryuk:0.3.0   "/app"      10 minutes ago   Up 10 minutes   0.0.0.0:55004->8080/tcp   testcontainers-ryuk-de1a8ac0-cebd-4cb2-a028-777e4331ec39
```

Now if I leave the java process running - I noticed that I got an additional stack trace when I restarted the docker process. That output is as follows:

```[main] DEBUG org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd - Cmd: db130b1922e2fb4c50775af81e9c6414e73be1d63db8cf8c4d31dce3931cf9d5,true,true
[main] DEBUG org.testcontainers.DockerClientFactory - Failure while checking for mountable file support
com.github.dockerjava.api.exception.DockerException: Status 502: Bad response from Docker engine

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:249)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.delete(DefaultInvocationBuilder.java:56)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.RemoveContainerCmdExec.execute(RemoveContainerCmdExec.java:28)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.RemoveContainerCmdExec.execute(RemoveContainerCmdExec.java:11)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.RemoveContainerCmdImpl.exec(RemoveContainerCmdImpl.java:67)
	at org.testcontainers.DockerClientFactory.runInsideDocker(DockerClientFactory.java:365)
	at org.testcontainers.DockerClientFactory.runInsideDocker(DockerClientFactory.java:347)
	at org.testcontainers.DockerClientFactory.checkMountableFile(DockerClientFactory.java:309)
	at org.testcontainers.DockerClientFactory.isFileMountingSupported(DockerClientFactory.java:84)
	at org.testcontainers.containers.GenericContainer.applyConfiguration(GenericContainer.java:750)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:359)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:325)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:323)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:311)
	at Main.main(Main.java:14)
        ⚠︎ Unable to mount a file from test host into a running container. This may be a misconfiguration or limitation of your Docker environment. Some features might not work.
[main] DEBUG org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd - Cmd: org.testcontainers.shaded.com.github.dockerjava.core.command.CreateContainerCmdImpl@4bc222e[name=<null>,hostName=<null>,domainName=<null>,user=<null>,attachStdin=<null>,attachStdout=<null>,attachStderr=<null>,portSpecs=<null>,tty=<null>,stdinOpen=<null>,stdInOnce=<null>,env={HOSTNAME_EXTERNAL=localhost,SERVICES=dynamodb},cmd={},healthcheck=<null>,argsEscaped=<null>,entrypoint=<null>,image=localstack/localstack:latest,volumes=com.github.dockerjava.api.model.Volumes@2dc9b0f5,workingDir=<null>,macAddress=<null>,onBuild=<null>,networkDisabled=<null>,exposedPorts=com.github.dockerjava.api.model.ExposedPorts@6531a794,stopSignal=<null>,stopTimeout=<null>,hostConfig=HostConfig(binds=[/var/run/docker.sock:/var/run/docker.sock:rw], blkioWeight=null, blkioWeightDevice=null, blkioDeviceReadBps=null, blkioDeviceWriteBps=null, blkioDeviceReadIOps=null, blkioDeviceWriteIOps=null, memorySwappiness=null, nanoCPUs=null, capAdd=null, capDrop=null, containerIDFile=null, cpuPeriod=null, cpuRealtimePeriod=null, cpuRealtimeRuntime=null, cpuShares=null, cpuQuota=null, cpusetCpus=null, cpusetMems=null, devices=null, deviceCgroupRules=null, deviceRequests=null, diskQuota=null, dns=null, dnsOptions=null, dnsSearch=null, extraHosts=[], groupAdd=null, ipcMode=null, cgroup=null, links=[], logConfig=LogConfig(type=null, config=null), lxcConf=null, memory=null, memorySwap=null, memoryReservation=null, kernelMemory=null, networkMode=null, oomKillDisable=null, init=null, autoRemove=null, oomScoreAdj=null, portBindings={}, privileged=null, publishAllPorts=true, readonlyRootfs=null, restartPolicy=null, ulimits=null, cpuCount=null, cpuPercent=null, ioMaximumIOps=null, ioMaximumBandwidth=null, volumesFrom=[], mounts=null, pidMode=null, isolation=null, securityOpts=null, storageOpt=null, cgroupParent=null, volumeDriver=null, shmSize=null, pidsLimit=null, runtime=null, tmpFs=null, utSMode=null, usernsMode=null, sysctls=null, consoleSize=null),labels={org.testcontainers=true, org.testcontainers.sessionId=64ab1b64-6322-425d-bf5d-5e84d11650bb},shell=<null>,networkingConfig=<null>,ipv4Address=<null>,ipv6Address=<null>,aliases=<null>,authConfig=AuthConfig(username=rmoak, email=null, registryAddress=index.docker.io, stackOrchestrator=null)]
[main] ERROR 🐳 [localstack/localstack:latest] - Could not start container
com.github.dockerjava.api.exception.DockerException: Status 502: Bad response from Docker engine

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:249)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:124)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:33)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:13)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.CreateContainerCmdImpl.exec(CreateContainerCmdImpl.java:595)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:407)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:325)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:323)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:311)
	at Main.main(Main.java:14)
Exception in thread "main" org.testcontainers.containers.ContainerLaunchException: Container startup failed
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:330)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:311)
	at Main.main(Main.java:14)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:323)
	... 2 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:497)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:325)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 3 more
Caused by: com.github.dockerjava.api.exception.DockerException: Status 502: Bad response from Docker engine

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:249)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:124)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:33)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:13)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.CreateContainerCmdImpl.exec(CreateContainerCmdImpl.java:595)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:407)
	... 5 more
```
	
Seeing the error: `Unable to mount a file from test host into a running container. This may be a misconfiguration or limitation of your Docker environment. Some features might not work.` caused me to poke around in the settings and I found that `Use gRPC FUSE for file sharing` option which was disabled.

If I disable the option, apply and restart then I can repro. 
If I enable the option, apply and restart then everything works.

I know I ran into some issue that caused me to disable that (unrelated to any of this) but I can't recall what it was now. I will leave it enabled to see if there are any further occurrences - but I seem to be stable. 

Thank you again.

**Comment by mgagliardo** [comment]
Glad you could solve it out @ryanmoaksonos . Thanks for replying.
